ersilia/core/tracking.py

```diff
@@ -110,38 +110,52 @@ def create_persistent_file(model_id):
     with open(file_name, "w") as f:
         f.write("Session started for model: {0}\n".format(model_id))
     
-    return file_name
 
+def check_file_exists(model_id):
+    """
+    Check if the persistent file exists.
+    :param model_id: The currently running model
+    :return: True if the file exists, False otherwise.
+    """
+    persistent_file_dir = os.path.join(EOS, ERSILIA_RUNS_FOLDER, "session", model_id)
+    file_name = os.path.join(persistent_file_dir, "current_session.txt")
+    return os.path.isfile(file_name)
     
+       
         
 def write_persistent_file(contents, model_id):
-
     """
     Writes contents to the current persistent file. Only writes if the file actually exists.
     :param contents: The contents to write to the file.
+    :param model_id: The currently running model
     """
-    
-    file_name = create_persistent_file(model_id)
-    if file_name and os.path.isfile(file_name):
+    if check_file_exists(model_id):
+        file_name = os.path.join(EOS, ERSILIA_RUNS_FOLDER, "session", model_id, "current_session.txt")
         with open(file_name, "a") as f:
             f.write(f"{contents}\n")
 
-
+    else:
+        raise FileNotFoundError(f"The persistent file for model {model_id} does not exist. Cannot write contents.")
```

Comment:
Great job adding the `check_file_exists()` function @Malikbadmus.

Here's my thought. Wouldn't it be better to create the file and write the content to it, if the file doesn't exist in the `else` block instead of raising an error?

Reply:
Many thanks @Inyrkz .

My thought was an error will be more appropriate, since it indicated something might have gone wrong or a process has been bypass, leading to a rerun of the processes. 

Write_persistent_file() is being used by many methods, adding create_file_path function in the else block might lead to a session file being created with insufficient information, or what do you think?

Reply:
I think I'm in the favor of raising an error rather than silently ignoring it so I am with @Malikbadmus on this one.